source/slang/slang-check-expr.cpp

```diff
@@ -3598,6 +3652,10 @@ namespace Slang
             if (!isTypePack(baseType) && !as<TupleType>(baseType))
                 goto error;
         }
+        
+        if (auto tupleType = as<TupleType>(baseType))
```

Comment:
Here we are trying to figure out the captured type pack from the inner type of a `each X` type. If `X` is a Tuple type, because we know a tuple type is represented as `TupleType(TypePack(...))`, we should instead use the inner type pack as the captured type pack argument for the enclosing `expand` type.